daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarLint Daemon
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.daemon.model;
+
+import org.sonarsource.sonarlint.core.client.api.common.LogOutput;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.LogEvent;
+
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+public class ProxyLogOutput implements LogOutput {
+  private StreamObserver<LogEvent> response;
+
+  public void setObserver(StreamObserver<LogEvent> response) {
+    if (this.response != null) {
+      this.response.onCompleted();
+    }
+    this.response = response;
+  }
+
+  @Override
+  public synchronized void log(String formattedMessage, Level level) {
+    if (level == Level.ERROR) {
+      System.err.println(formattedMessage);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)